fake_filesystem_test.py

```diff
@@ -3279,6 +3279,10 @@ def testCharactersBeforeRootIgnoredInJoinPaths(self):
   def testResolvePath(self):
     self.assertEqual('c:/foo/bar', self.filesystem.ResolvePath('c:/foo/bar'))
 
+  def testGetPathComponents(self):
+    self.assertEqual(['c:', 'foo', 'bar'], self.filesystem.GetPathComponents('c:/foo/bar'))
+    self.assertEqual(['c:'], self.filesystem.GetPathComponents('c:'))
+
```

Comment:
Is there any kind of degenerate but still legal path that should be tried, such as `c:foo/bar` or `c:\foo/bar`?  I do not know if `c:foo/bar` is in fact legal.